eis_toolkit/prediction/mlp.py

```diff
@@ -0,0 +1,368 @@
+    specified, a Dropout layer is added after each Dense layer. In the end, a Flatten layer is added followed by
+    the output layer (Dense that uses `output_neruons` and `last_activation`).
+
+    Args:
+        X: Input data.
+        y: Target labels.
+        neurons: Number of neurons in each hidden layer. Defaults to one layer with 16 neurons.
+        validation_split: Fraction of data used for validation during training. Value must be > 0 and < 1 or None.
+            Defaults to 0.2.
+        activation: Activation function used in each hidden layer. Defaults to 'relu'.
+        output_neurons: Number of neurons in the output layer. Defaults to 1.
+        last_activation: Activation function used in the output layer. Defaults to 'linear'.
+        epochs: Number of epochs to train the model. Defaults to 50.
+        batch_size: Number of samples per gradient update. Defaults to 32.
+        optimizer: Optimizer to be used. Defaults to 'adam'.
+        learning_rate: Learning rate to be used in training. Value must be > 0. Defalts to 0.001.
+        loss_function: Loss function to be used. Defaults to 'mse'.
+        dropout_rate: Fraction of the input units to drop. Value must be >= 0 and <= 1. Defaults to None.
+        early_stopping: Whether or not to use early stopping in training. Defaults to True.
+        es_patience: Number of epochs with no improvement after which training will be stopped. Defaults to 5.
+        metrics: Metrics to be evaluated by the model during training and testing. Defaults to ['mse'].
+
+    Returns:
+        Trained MLP model and training history.
+
+    Raises:
+        InvalidParameterValueException: Some of the numeric parameters have invalid values.
+    """
+    # 1. Check input data
+    _check_MLP_inputs(
+        neurons=neurons,
+        validation_split=validation_split,
+        learning_rate=learning_rate,
+        dropout_rate=dropout_rate,
+        es_patience=es_patience,
+        batch_size=batch_size,
+        epochs=epochs,
+        output_neurons=output_neurons,
+        loss_function=loss_function,
+    )
+
+    # 2. Create and compile a sequential model
+    model = keras.Sequential()
+
+    for neuron in neurons:
+        model.add(keras.layers.Dense(units=neuron, activation=activation))
+
+        if dropout_rate is not None:
+            model.add(keras.layers.Dropout(dropout_rate))
+
+    model.add(keras.layers.Flatten())
+
+    model.add(keras.layers.Dense(units=output_neurons, activation=last_activation))
+
+    model.compile(
+        optimizer=_keras_optimizer(optimizer, learning_rate=learning_rate), loss=loss_function, metrics=metrics
+    )
+
+    # 3. Train the model
+    # Early stopping callback
+    callbacks = [keras.callbacks.EarlyStopping(monitor="val_loss", patience=es_patience)] if early_stopping else []
+
+    history = model.fit(
+        X,
+        y,
+        epochs=epochs,
+        validation_split=validation_split if validation_split else 0,
```

Comment:
Yeah good idea. I added this too. Do you think we want to include this option to provide already split validation data for the Sklearn ML models too?

Reply:
Yes, I was just thinking that we could provide an option in the functions we offer to give a separate validation dataset. When defined, it would override calling the `split_data` inside  `_train_and_validate_sklearn_model` and directly assign `X_valid` and `y_valid` from the given dataset. But I don't see a need to add this, just thought I'd ask. Anyway it is separate from this MLP issue.